src/analyzer/analyzerkey.h

```diff
@@ -6,19 +6,19 @@
 
 #include "analyzer/analyzer.h"
 #include "analyzer/plugins/analyzerplugin.h"
+#include "analyzer/plugins/analyzerpluginsupport.h"
 #include "preferences/keydetectionsettings.h"
 #include "preferences/usersettings.h"
 #include "track/track_decl.h"
 #include "util/memory.h"
 
-class AnalyzerKey : public Analyzer {
+QList<mixxx::AnalyzerPluginInfo> defaultAvailableKeyPlugins();
+
+class AnalyzerKey : private AnalyzerPluginSupportInfo, public Analyzer {
```

Comment:
using private inheritance as a means of code reuse seems very bad practice to me tbh.

Reply:
`AnalyzerPluginSupportInfo` is an interface. The simple interface design came about after seeing duplication of code between the (currently) two classes `Analyzer{Beats,Key}`. The only classes needing to use this interface and access any member-variables within, are `Analyzer{Beats,Key}`, _and no other external translation-units,_ that is why the interface `AnalyzerPluginSupportInfo` is declared `private` (for them).

The bonus of consolidating the common code into an interface-style is the code reduction (and natural beneficial code-reuse).

Some of this approach is explained in these articles:
* https://www.stroustrup.com/bs_faq2.html#abstract-class
* https://www.artima.com/articles/the-c-style-sweet-spot

I am happy to declare it something other than `private`, but as no other translation-unit is using the interface, it would seem slightly unorthodox.